code-challenges.test.js

```diff
@@ -1,52 +1,109 @@
-
-// Use test driven development to complete the following questions
-// Add appropriate dependencies: $ yarn add jest
-
-// Reminder: The test will call your function
-// Run the file with the following command: $ yarn jest
-
-// --------------------1) Create a function that takes in a string and returns a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
-
-// a) Create a test with expect statements using the variables provided.
-
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
-
-// b) Create the function that makes the test pass.
-
-// --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
-
-// a) Create a test with expects statement using the variable provided.
-
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
-
-const filterLetterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const filterLetterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
-
-// b) Create the function that makes the test pass.
-
-// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
-
-// a) Create a test with expect statements using the variable provided.
-
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
-
-// b) Create the function that makes the test pass.
+// --------------------1) 
+// Pseudo Code
+// Input: a string of word(s)
+// Output: returns a string with a coded message
+// - Declare a variable to map a/e/i/o to their encoded counterparts 4/3/2/1
+// - Create an array of encoded characters by iterating over each character of the input
+// - In the iteration, each character should be converted to lowercase and checks if there's a character that needs to be converted
+// - If there is, add the encoded value to a separate array, else it's just the original character added
+// - Use the join method to connect the characters inside the array into a single string
+
+const encoded = (string) => {
+    const code = {
+        "a": "4",
+        "e": "3",
+        "i": "1",
+        "o": "0"
+    }
+
+    const encodedChars = Array.from(string, (char) => {
+        const lowerCaseChar = char.toLowerCase()
+        return code[lowerCaseChar] || char
+    })
+
+    return encodedChars.join("")
+}
+describe("encoded", () => {
+
+    it("should encode 'Lackadaisical' to 'L4ck4d41s1c4l'", () => {
```

Comment:
Don't need 3 'it' blocks, just 1. Then you can add 3 'expect' statements in that 1 'it' statement.